### PR TITLE
AAP-27696 Create Using Ansible Plug-ins guide

### PR DESCRIPTION
**Pull request by ariordan-redhat**
[AAP-27696](https://issues.redhat.com/browse/AAP-27696) Add Using Ansible plug-in doc.

[Preview build](https://ariordan-redhat.github.io/aap-builds/aap-plugin-rhdh-using-aap-27696-2024-07-26.html)

Affects `titles/aap-plugin-rhdh-using`

### CODE REVIEW

File: downstream/assemblies/devtools/assembly-rhdh-using.adoc [hunk 1]
```diff
@@ -1,14 +1,25 @@
...
-= Using {AAPRHDH}
+= Using the Ansible plug-ins
 
 :context: rhdh-using
 [role="_abstract"]
+You can use Ansible plug-ins for {RHDH} (RHDH) to learn about Ansible, create automation projects, and access opinionated workflows and tools to develop and test your automation code.
+From the {RHDH} UI, you can navigate to your {PlatformNameShort} instance, where you can configure and run automation jobs. 
 
-Using the plug-in - placeholder
+This document describes  how to use the Ansible plug-ins for {RHDH}.
```
Comment (EMcWhinn) [inline downstream/assemblies/devtools/assembly-rhdh-using.adoc:11]:
```suggestion
This document describes how to use the Ansible plug-ins for {RHDH}.
```

File: downstream/modules/devtools/proc-rhdh-create.adoc [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+Ask your administrator to assign access to you if necessary.
+
+.Procedure:
+
+. Log in to your {RHDH} UI.
+. Click the Ansible `A` icon in the {RHDH} navigation panel.
+. Navigate to the *Overview* page.
+. Click *Create*.
+. Click *Create Ansible Git Project*.
+. The *Available Templates* page opens.
```
Comment (EMcWhinn) [inline downstream/modules/devtools/proc-rhdh-create.adoc:18]:
```suggestion
The *Available Templates* page opens.
```
I would remove this as a step as no action is completed by the user.

File: downstream/modules/devtools/proc-rhdh-create.adoc [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+{HubNameStart} stores existing collections and execution environments that you can use in your projects.
+For more information, see
+xref:rhdh-discover-collections_rhdh-using[Discovering existing Ansible content].
+* *Create* creates new projects in your configured Source Control Management platforms such as GitHub.
+For more information, see the Creating a project section.
+* *Develop* links you to OpenShift Dev Spaces, if configured in the Ansible plug-ins installation.
+OpenShift Dev Spaces provides on-demand, web-based Integrated Development Environments (IDEs), where you can develop automation content.
+For more information, see the Developing projects section.
+* *Operate* connects you to {PlatformNameShort}, where you can create and run automation jobs that use the projects you have developed. 
+For more information, see the Operating automation tasks section.
```
Comment (EMcWhinn) [inline downstream/modules/devtools/proc-rhdh-create.adoc:75]: May have missed this but is there an xref for the **Operating automation tasks** section?

File: downstream/modules/devtools/proc-rhdh-execute-automation-devspaces.adoc [hunk 4]
```diff
@@ -0,0 +1,12 @@
+:_mod-docs-content-type: PROCEDURE
+
+[id="rhdh-execute-automation-devspaces_{context}"]
+= Executing automation tasks in Dev Spaces
+
+The default configuration for Dev Spaces provides access to the Ansible command line tools. 
+
+To execute an automation task in Dev Spaces from the VSCode user interface,
+right-click on a playbook name in the list of files and select *Run Ansible Playbook via ansible-navigator run* or *Run playbook via ansible-playbook*.
```
Comment (EMcWhinn) [inline downstream/modules/devtools/proc-rhdh-execute-automation-devspaces.adoc:9]:
```suggestion
right-click a playbook name in the list of files and select *Run Ansible Playbook via ansible-navigator run* or *Run playbook via ansible-playbook*.
```

I'd avoid the use of the preposition on with mouse actions, as per SG.

File: downstream/modules/devtools/proc-rhdh-firewall-example-discover.adoc [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+{HubNameStart} is a central repository for discovering, downloading, and managing trusted content collections from Red Hat and its partners.
+{PrivateHubNameStart} provides an on-premise solution for managing content collections.
+
+. Click on the Ansible `A` icon in the {RHDH} navigation panel.
+. Click *Discover existing collections*.
+. Click *Go to Automation Hub*. 
++
+--
+** If {PrivateHubName} has been configured in the Ansible plug-ins, you will be redirected to your *PrivateHubName* instance.
```
Comment (EMcWhinn) [inline downstream/modules/devtools/proc-rhdh-firewall-example-discover.adoc:16]:
```suggestion
** If {PrivateHubName} has been configured in the Ansible plug-ins, you are redirected to your *PrivateHubName* instance.
```

File: downstream/modules/devtools/proc-rhdh-firewall-example-discover.adoc [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+** If {PrivateHubName} has not been configured in the Ansible plug-ins installation configuration,
+you will be redirected to the Red Hat Hybrid Console (RHCC) automation hub.
```
Comment (EMcWhinn) [inline downstream/modules/devtools/proc-rhdh-firewall-example-discover.adoc:18]:
```suggestion
you are redirected to the Red Hat Hybrid Console (RHCC) automation hub.
```

File: downstream/modules/devtools/proc-rhdh-firewall-example-edit.adoc [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+The Ansible plug-ins integrate OpenShift Dev Spaces to edit your Ansible projects.
+OpenShift Dev Spaces provides on-demand, web-based Integrated Development Environments (IDEs).
+
+Ansible Git projects provisioned using the Ansible plug-ins include best practice configurations for OpenShift Dev Spaces.
+These configurations include installing the Ansible VS Code extension and providing access from the IDE terminal to Ansible development tools,
+such as Ansible Navigator and Ansible Lint.
+
+[NOTE]
+====
+OpenShift DevSpaces is optional and it is not required to run the Ansible plug-ins.
```
Comment (EMcWhinn) [inline downstream/modules/devtools/proc-rhdh-firewall-example-edit.adoc:15]: I noticed there is no space between Dev and Spaces here but there is in other mentions of it, should it be consistent?

File: downstream/modules/devtools/proc-rhdh-firewall-example-edit.adoc [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+====
+
+This example assumes that OpenShift Dev Spaces has been configured in the Ansible plug-ins installation.
+
+.Procedure
+
+* In the *catalog item* view of your playbook project, click *Open Ansible project in OpenShift Dev Spaces*.
++
+A VS Code instance of OpenShift Dev Spaces opens in a new browser tab.
+It automatically loads your new Ansible Playbook git project.
```
Comment (EMcWhinn) [inline downstream/modules/devtools/proc-rhdh-firewall-example-edit.adoc:26]:
```suggestion
It automatically loads your new Ansible Playbook Git project.
```

File: downstream/modules/devtools/proc-rhdh-firewall-example-new-playbook.adoc [hunk 9]
```diff
@@ -0,0 +1,41 @@
+:_mod-docs-content-type: PROCEDURE
+
+[id="rhdh-firewall-example-new-playbook_{context}"]
+= Creating a new playbook to automate the firewall configuration
+
+Create a new playbook and use the RHEL System Role collection to automate your Red Hat Enterprise Linux firewall configuration.
```
Comment (EMcWhinn) [inline downstream/modules/devtools/proc-rhdh-firewall-example-new-playbook.adoc:6]:
```suggestion
Create a new playbook and use the RHEL System Role collection to automate your {RHEL} firewall configuration.
```

File: downstream/modules/devtools/proc-rhdh-set-up-controller-project.adoc [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+Alternatively, if your platform instance was not configured during the Ansible plug-in installation, navigate to your {ControllerName} instance in a browser and log in.
+. Log in to {ControllerName}.
+. Create a project in {PlatformNameShort} for the GitHub repository where you stored your playbook project.
+Refer to the
+link:{BaseURL}/red_hat_ansible_automation_platform/{PlatformVers}/html/automation_controller_user_guide/index#controller-projects[Projects]
+chapter of the _Automation controller user guide_.
+. Create a job template that uses a playbook from the project that you created.
+Refer to the
+link:{BaseURL}/red_hat_ansible_automation_platform/{PlatformVers}/html/automation_controller_user_guide/index#controller-create-job-template[Job Templates]
+chapter of the _Automation controller user guide_.
```
Comment (EMcWhinn) [inline downstream/modules/devtools/proc-rhdh-set-up-controller-project.adoc:25]:
```suggestion
chapter of the _Using automation execution_.
```

File: downstream/modules/devtools/proc-rhdh-set-up-controller-project.adoc [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
++
+Alternatively, if your platform instance was not configured during the Ansible plug-in installation, navigate to your {ControllerName} instance in a browser and log in.
+. Log in to {ControllerName}.
+. Create a project in {PlatformNameShort} for the GitHub repository where you stored your playbook project.
+Refer to the
+link:{BaseURL}/red_hat_ansible_automation_platform/{PlatformVers}/html/automation_controller_user_guide/index#controller-projects[Projects]
+chapter of the _Automation controller user guide_.
+. Create a job template that uses a playbook from the project that you created.
+Refer to the
+link:{BaseURL}/red_hat_ansible_automation_platform/{PlatformVers}/html/automation_controller_user_guide/index#controller-create-job-template[Job Templates]
```
Comment (EMcWhinn) [inline downstream/modules/devtools/proc-rhdh-set-up-controller-project.adoc:24]:
```suggestion
link:{BaseURL}/red_hat_ansible_automation_platform/{PlatformVers}/html/using_automation_execution/index#controller-job-templates[Job templates]
```

File: downstream/modules/devtools/ref-rhdh-discover-collections.adoc [hunk 12]
```diff
@@ -0,0 +1,14 @@
+:_mod-docs-content-type: REFERENCE
+
+[id="rhdh-discover-collections_{context}"]
+= Discovering collections
+
+From the *Overview* page in the Ansible plug-ins dashboard on {RHDH}, click *Discover Existing Collections*.
+
+The links in this pane provide access to the source of reusable automation content collections that you configured during plug-in installation. 
+
+If you configured {PrivateHubName} when installing the plug-in, you can click *Go to Automation Hub* to view the collections and execution environments that your enterprise has curated.
```
Comment (EMcWhinn) [inline downstream/modules/devtools/ref-rhdh-discover-collections.adoc:10]:
```suggestion
If you configured {PrivateHubName} when installing the plug-in, you can click *Go to Automation Hub* to view the collections and {ExecEnvShort}s that your enterprise has curated. 
```

**Review by EMcWhinn — COMMENTED** [review]
Some suggestions left.

File: downstream/modules/devtools/proc-rhdh-set-up-controller-project.adoc (hunk 11 above)
Comment (ariordan-redhat) [reply downstream/modules/devtools/proc-rhdh-set-up-controller-project.adoc:24]: @EMcWhinn Is this for 2.5?
The content needs to go in 2.4 also

**Comment by g-murray** [comment]
LGTM

File: downstream/modules/devtools/proc-rhdh-set-up-controller-project.adoc (hunk 10 above)
Comment (ariordan-redhat) [reply downstream/modules/devtools/proc-rhdh-set-up-controller-project.adoc:27]: Will need to change these for 2.5

**Comment by ariordan-redhat** [comment]
Need to update some links in main before I can backport to 2.5

**Pull request merged**